File: app/src/main/java/com/u1fukui/bbs/view/customview/ObservableListRecyclerAdapter.java
package com.u1fukui.bbs.view.customview;

import android.databinding.ObservableList;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;

public abstract class ObservableListRecyclerAdapter<T, VH extends RecyclerView.ViewHolder>
        extends ArrayRecyclerAdapter<T, VH> {

    public ObservableListRecyclerAdapter(@NonNull ObservableList<T> list) {
        super(list);

        list.addOnListChangedCallback(new ObservableList.OnListChangedCallback<ObservableList<T>>() {
            @Override
            public void onChanged(ObservableList<T> contributorViewModels) {
                notifyDataSetChanged();
            }

            @Override
            public void onItemRangeChanged(ObservableList<T> contributorViewModels, int i, int i1) {
                notifyItemRangeChanged(i, i1);
            }

            @Override
            public void onItemRangeInserted(ObservableList<T> contributorViewModels, int i, int i1) {
                notifyItemRangeInserted(i, i1);
            }

            @Override
            public void onItemRangeMoved(ObservableList<T> contributorViewModels, int i, int i1,
                    int i2) {
                notifyItemMoved(i, i1);
            }

            @Override
            public void onItemRangeRemoved(ObservableList<T> contributorViewModels, int i, int i1) {
                notifyItemRangeRemoved(i, i1);
            }
        });
    }
}

File: app/src/main/java/com/u1fukui/bbs/customview/ObservableListRecyclerAdapter.kt
package com.u1fukui.bbs.customview

import android.databinding.ObservableList
import android.support.v7.widget.RecyclerView

abstract class ObservableListRecyclerAdapter<T, VH : RecyclerView.ViewHolder>(list: ObservableList<T>) :
    ArrayRecyclerAdapter<T, VH>(list) {

    init {
        list.addOnListChangedCallback(object :
            ObservableList.OnListChangedCallback<ObservableList<T>>() {
            override fun onChanged(list: ObservableList<T>) {
                notifyDataSetChanged()
            }

            override fun onItemRangeChanged(list: ObservableList<T>, i: Int, i1: Int) {
                notifyItemRangeChanged(i, i1)
            }

            override fun onItemRangeInserted(list: ObservableList<T>, i: Int, i1: Int) {
                notifyItemRangeInserted(i, i1)
            }

            override fun onItemRangeMoved(list: ObservableList<T>, i: Int, i1: Int, i2: Int) {
                notifyItemMoved(i, i1)
            }

            override fun onItemRangeRemoved(list: ObservableList<T>, i: Int, i1: Int) {
                notifyItemRangeRemoved(i, i1)
            }
        })
    }
}
